toolchain/parse/testdata/function/declaration/basic.carbon

```diff
@@ -8,8 +8,153 @@
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/parse/testdata/function/declaration/basic.carbon
 
```

Comment:
Should we maybe get rid of the whole `declaration` directory, and move this file to `function/declaration.carbon`?

Reply:
Sure, I'd been on the fence about it but not sure we'd have split dirs if we had split files before. Also merged the two extern.carbon's this way.